lib/tizen_cache.dart

```diff
@@ -121,18 +137,40 @@ class TizenEngineArtifacts extends EngineCachedArtifact {
   @override
   List<String> getPackageDirs() => const <String>[];
 
+  @override
+  bool isUpToDateInner(FileSystem fileSystem) {
+    // Download always happens, if following variables are set.
+    if (_platform.environment['TIZEN_ENGINE_GITHUB_RUN_ID'] != null ||
+        _platform.environment['TIZEN_ENGINE_AZURE_BUILD_ID'] != null) {
+      return false;
+    }
+    return super.isUpToDateInner(fileSystem);
+  }
+
   /// See: [EngineCachedArtifact.updateInner] in `cache.dart`
   @override
   Future<void> updateInner(
     ArtifactUpdater artifactUpdater,
     FileSystem fileSystem,
     OperatingSystemUtils operatingSystemUtils,
   ) async {
-    final String? buildId = _platform.environment['AZURE_BUILD_ID'];
-    final String downloadUrl = buildId == null
-        ? '$engineBaseUrl/download/$shortVersion'
-        : await _getDownloadUrlFromAzure(buildId);
+    final String? ghRunId = _platform.environment['TIZEN_ENGINE_GITHUB_RUN_ID'];
```

Comment:
`GITHUB_RUN_ID` is used in Github Actions's build process. https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables
So. I added the prefix `TIZEN_ENGINE_` to avoid the conflict.

Reply:
Isn't the variable only used in workflows?

Reply:
Yes, in case of building an app in workflow, the build will probably fail due to the wrong `GITHUB_RUN_ID` value.

Reply:
how about `ENGINE_GITHUB_RUN_ID` and `ENGINE_AZURE_BUILD_ID`?

Reply:
Do you have an example of the failing case? As far as I know the workflows are run by GitHub Actions workers and the `TIZEN_ENGINE_GITHUB_RUN_ID` variable is set by users during development, so there's no chance of confliction being made between them.

Reply:
If we change the variable name to `GITHUB_RUN_ID`, `flutter-tizen` will try to download the artifacts from Github Actions with the `GITHUB_RUN_ID` set by the Github Actions worker when someone try building an app with `flutter-tizen build tpk` in Github Actions workflow.
```yaml
// GITHUB_RUN_ID is set by Github Actions worker to indicate current running RUN_ID.
// following command will fail because it will try to download artifacts from current running action run.
- run: flutter-tizen build tpk -pmobile 
```

Reply:
Ah, now I understand what you mean. It may affect the plugins CI build.

In that case `TIZEN_ENGINE_GITHUB_RUN_ID` should be just fine.

Reply:
Didn't you miss the variable naming issue?

Reply:
oops sorry, I missed..